doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -1,89 +1,61 @@
 
-To focus our attention we'll analyze `mod_offline` which is responsible
-for storing messages for delivery to users unavailable at the time
-of sending.
+To focus our attention, we'll analyze `mod_offline` which is responsible for storing messages for delivery to users unavailable at the time of sending.
 `mod_offline` is an implementation of [XEP-0203][xep-0203].
 
 [xep-0203]: http://xmpp.org/extensions/xep-0203.html
 
 ## Running a hook
 
-`ejabberd_sm` (ejabberd/MongooseIM session manager) is the module which discovers
-whether the recipient of a message is available or not.
-That's where storage of the message for later delivery takes place.
-The simplest way of calling into `mod_offline` to store a message
-would be to make a call like:
+`ejabberd_sm` (ejabberd/MongooseIM session manager) is the module which discovers whether the recipient of a message is available or not.
+That's where storing the message for later delivery takes place.
+Here's a simple example of calling `mod_offline`:
 
 ```erlang
 mod_offline:store_packet(From, To, Packet)
 ```
 
-However, that would couple `ejabberd_sm` with `mod_offline`.
-I.e. if `mod_offline` was not available the code would simply crash;
-if it was misconfigured or turned off the behaviour would be undefined.
-To avoid that coupling and also to enable other
-([possibly yet to be written](#sidenote-yet-to-be-written))
-code to carry out some action at this particular
-moment of the message processing pipeline `ejabberd_sm` instead calls:
+Note that in this example `ejabberd_sm` is coupled with `mod_offline`.
+I.e. if `mod_offline` was not available, the code would simply crash; if it was misconfigured or turned off, the behaviour would be undefined.
+To avoid that coupling and also to enable other ([possibly yet to be written](#sidenote-yet-to-be-written)) code to carry out some action at this particular moment, `ejabberd_sm` instead calls:
 
 ```erlang
 ejabberd_hooks:run(offline_message_hook, LServer,
                    [From, To, Packet])
 ```
 
-The extra level of indirection introduced by this call gives the flexibility
-to determine at runtime what code actually gets run at this point.
-`offline_message_hook` is just the name of the hook (in other words of
-the event that is being signalled); `From`, `To` and `Packet` are the
-arguments passed to the handler just as they would in case of the
-function being called directly; `LServer` is
-[the XMPP domain for which this hook is signalled](#sidenote-multiple-domains).
+The extra level of indirection introduced by this call gives the flexibility to determine at runtime what code actually gets run at this point.
+`offline_message_hook` is just the name of the hook (in other words of the event that is being signalled); `From`, `To` and `Packet` are the arguments passed to the handler just as they would in case of the function being called directly; `LServer` is [the XMPP domain for which this hook is signalled](#sidenote-multiple-domains).
```

Comment:
I suggest breaking the line on `;` as well. This particular line is far too long.